src/components/displaySvg/finalSvgDisplay.js

```diff
@@ -0,0 +1,154 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import finalSvgDisplay from "./finalSvgDisplay.scss";
+import { connect } from "react-redux";
+import { SVG_TAG_NAMES } from "../../constants/constants";
+import {
+  ADD_VARIABLE,
+  UPDATE_VARIABLE,
+  DELETE_VARIABLE
+} from "../../store/actionTypes";
+import { wrapElements, updateSvgElements } from "../functions";
+
+class FinalSvgDisplay extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      wrappedPathsElement: [],
+      pathArrayState: null,
+      storePathsIds: [],
+      clickedSvgState: []
+    };
+  }
+
+  downloadOptimizedSvgs = () => {
+    
+    let svgElementList = document.getElementsByTagName("svg");
+
+    //get svg source.
+    let serializer = new XMLSerializer();
+    for(let svgDataIndex in svgElementList){   
+    //get svg source.
+    let source = serializer.serializeToString(svgElementList[svgDataIndex]);
+    source = source.replace("<a class=\"svg-elem-wrapper\" id=\"id-1\">","").replace("</a>","").replace("onclick=\"changeSvgColourById(this.id);\"","");
+
+    //add name spaces.
+    if (!source.match(/^<svg[^>]+xmlns="http\:\/\/www\.w3\.org\/2000\/svg"/)) {
+      source = source.replace(
+        /^<svg/,
+        '<svg xmlns="http://www.w3.org/2000/svg"'
+      );
+    
+    }
+    if (!source.match(/^<svg[^>]+"http\:\/\/www\.w3\.org\/1999\/xlink"/)) {
+      source = source.replace(
+        /^<svg/,
+        '<svg xmlns:xlink="http://www.w3.org/1999/xlink"'
+      );
+    }
+
+    //add xml declaration
+    source = '<?xml version="1.0" standalone="no"?>\r\n' + source;
+
+    //convert svg source to URI data scheme.
+    let url = "data:image/svg+xml;charset=utf-8," + encodeURIComponent(source);
+    //set url value to a element's href attribute.
+        let dl = document.createElement("a");
+        document.body.appendChild(dl); // This line makes it work in Firefox.
+        dl.setAttribute("href", url);
+        dl.setAttribute("download", "test.svg");
+        dl.click();
+
+      if(svgElementList.length-1 == svgDataIndex){
+        break;
+      }
+    }
+
+  };
+
+  componentDidMount = () => {
+    let array = [];
+    let pathArrayLocal = [];
+    let elementIds = [];
+    wrapElements(SVG_TAG_NAMES, array, pathArrayLocal, 7, elementIds);
+    let functionStr =
```

Comment:
rather inject this function directly into the SVG than inject it to the DOM and then inject it to the svg file

Reply:
I cant actually , figure out how to inject it directly into svg, as with the options I tried its either element id is undefined or cant find the function its self, and when use `elem.setAttribute('onClick', function(){
// code here
});`

with the above approach its expose my code in the element and does nothing after all.